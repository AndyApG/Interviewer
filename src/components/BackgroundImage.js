import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { assetUrl } from '../utils/protocol';

class BackgroundImage extends Component {
  static propTypes = {
    path: PropTypes.string.isRequired,
    style: PropTypes.object,
    getAssetUrl: PropTypes.func.isRequired,
  };

  static defaultProps = {
    style: {},
  };

  constructor(props) {
    super(props);

    this.state = {
      src: '',
    };
  }

  componentDidMount() {
    const { getAssetUrl, path } = this.props;
    getAssetUrl(path).then(src => this.setState({ src }));
  }

  render() {
    const { getAssetUrl, path, style, ...props } = this.props;
    return (
      <div
        style={{ ...style, backgroundImage: `url(${this.state.src})` }}
        {...props}
      />
    );
  }
}

const mapStateToProps = state => ({
  getAssetUrl: assetPath => assetUrl(state.protocol.path, assetPath),
});

export { BackgroundImage };

export default connect(mapStateToProps)(BackgroundImage);
