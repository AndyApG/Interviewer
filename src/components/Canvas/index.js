/* eslint-disable import/prefer-default-export */

export { default as EdgeLayout } from './EdgeLayout';
