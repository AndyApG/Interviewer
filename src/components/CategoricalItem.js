import React from 'react';
import { compose, withProps, withState } from 'recompose';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Flipped } from 'react-flip-toolkit';
import ReactMarkdown from 'react-markdown';
import { ALLOWED_MARKDOWN_LABEL_TAGS } from '@codaco/ui/src/utils/config';

import { DropTarget, MonitorDropTarget } from '../behaviours/DragAndDrop';
import { NodeList } from './';

/**
  * Renders a droppable CategoricalBin item
  */
const CategoricalItem = ({
  accentColor,
  details,
  id,
  isExpanded,
  isOver,
  label,
  nodes,
  onClick,
  onClickItem,
  sortOrder,
  willAccept,
}) => {
  const classNames = cx(
    'categorical-item',
    { 'categorical-item--hover': willAccept && isOver },
    { 'categorical-item--expanded': isExpanded },
  );

  return (
    <Flipped flipId={id}>
      <div
        className={classNames}
        style={{ '--categorical-item-color': accentColor }}
        onClick={onClick}
      >
        <div className="categorical-item__disk" />
        <div className="categorical-item__inner">
          <Flipped inverseFlipId={id} scale>
            <div className="categorical-item__title">
              <h3>
                <ReactMarkdown source={label} allowedTypes={ALLOWED_MARKDOWN_LABEL_TAGS} />
              </h3>
              {(!isExpanded && details) &&
                <h5>
                  <ReactMarkdown source={details} allowedTypes={ALLOWED_MARKDOWN_LABEL_TAGS} />
                </h5>}
            </div>
          </Flipped>
          {isExpanded &&
            <div className="categorical-item__content">
              <NodeList
                listId={`CATBIN_NODE_LIST_${label}`}
                id={`CATBIN_NODE_LIST_${label}`}
                onItemClick={onClickItem}
                items={nodes}
                sortOrder={sortOrder}
              />
            </div>
          }
        </div>
      </div>
    </Flipped>
  );
};

CategoricalItem.propTypes = {
  accentColor: PropTypes.string,
  details: PropTypes.string,
  id: PropTypes.string.isRequired,
  isExpanded: PropTypes.bool,
  isOver: PropTypes.bool,
  label: PropTypes.string,
  nodes: PropTypes.array,
  onClick: PropTypes.func,
  onClickItem: PropTypes.func,
  sortOrder: PropTypes.array,
  willAccept: PropTypes.bool,
};

CategoricalItem.defaultProps = {
  accentColor: 'black',
  details: '',
  isExpanded: false,
  isOver: false,
  label: 'undefined',
  nodes: [],
  onClick: () => {},
  onClickItem: () => {},
  sortOrder: [],
  willAccept: false,
};

export { CategoricalItem as UnconnectedCategoricalItem };

export default compose(
  withState('recentNode', 'setRecentNode', {}),
  withProps(props => ({
    accepts: () => true,
    onDrop: ({ meta }) => {
      props.onDrop({ meta });
      props.setRecentNode(meta);
    },
  })),
  DropTarget,
  MonitorDropTarget(['isOver', 'willAccept']),
)(CategoricalItem);
