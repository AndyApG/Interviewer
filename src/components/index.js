/* eslint-disable import/prefer-default-export */

export { default as Menu } from './Menu';
export { default as Scroller } from './Scroller';
