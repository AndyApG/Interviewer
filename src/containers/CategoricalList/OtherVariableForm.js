import React from 'react';
import PropTypes from 'prop-types';
import { reduxForm } from 'redux-form';
import { Button, Node } from '@codaco/ui';
import Field from '../Field';
import './OtherVariableForm.scss';

const stopClickPropagation = e =>
  e.stopPropagation();

const OtherVariableForm = ({
  otherVariableLabel,
  handleSubmit,
  color,
  label,
  onCancel,
}) => (
  <div className="other-variable-form" onClick={stopClickPropagation}>
    <form onSubmit={handleSubmit}>
      <div className="other-variable-form__header">
        <h2>Set {otherVariableLabel}</h2>
      </div>

      <div className="other-variable-form__content">
        <div className="other-variable-form__content-left">
          <Node label={label} color={color} />
        </div>
        <div className="other-variable-form__content-right">
          <Field
            label={otherVariableLabel}
            component="Text"
            name="otherVariable"
            validation={{ required: true }}
          />
        </div>
      </div>
      <div className="other-variable-form__footer">
        <Button type="button" color="white" onClick={onCancel}>Cancel</Button>
        <Button type="submit" aria-label="Submit" >Set {otherVariableLabel}</Button>
      </div>
    </form>
  </div>
);

OtherVariableForm.propTypes = {
  otherVariableLabel: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  color: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default reduxForm({ form: 'otherVariableForm' })(OtherVariableForm);
