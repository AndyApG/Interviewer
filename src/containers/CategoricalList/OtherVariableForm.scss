.other-variable-form {
  height: auto;

  &__header {
    margin-bottom: 2rem;
  }

  &__content {
    display: flex;
    width: 100%;
    align-items: center;

    &-left {
      flex: 0 0 auto;
      padding-bottom: 2rem;
    }

    &-right {
      flex: 1 1 auto;
      margin-left: 2rem;
    }
  }

  &__footer {
    text-align: right;

    button {
      margin-left: 1rem;
    }
  }
}
