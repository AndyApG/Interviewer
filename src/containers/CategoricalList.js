import React, { Component } from 'react';
import { compose, bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Flipper } from 'react-flip-toolkit';
import { find } from 'lodash';
import cx from 'classnames';

import { makeNetworkNodesForType, makeGetVariableOptions, makeGetPromptVariable } from '../selectors/interface';
import { actionCreators as sessionsActions } from '../ducks/modules/sessions';
import { CategoricalItem } from '../components/';
import { MonitorDragSource } from '../behaviours/DragAndDrop';
import { getCSSVariableAsString } from '../utils/CSSVariables';
import { getNodeAttributes, nodeAttributesProperty, nodePrimaryKeyProperty } from '../ducks/modules/network';

const colorPresets = [
  getCSSVariableAsString('--cat-color-seq-1'),
  getCSSVariableAsString('--cat-color-seq-2'),
  getCSSVariableAsString('--cat-color-seq-3'),
  getCSSVariableAsString('--cat-color-seq-4'),
  getCSSVariableAsString('--cat-color-seq-5'),
  getCSSVariableAsString('--cat-color-seq-6'),
  getCSSVariableAsString('--cat-color-seq-7'),
  getCSSVariableAsString('--cat-color-seq-8'),
  getCSSVariableAsString('--cat-color-seq-9'),
  getCSSVariableAsString('--cat-color-seq-10'),
];

const getCatColor = (itemNumber) => {
  if (itemNumber > 0) { return colorPresets[itemNumber % colorPresets.length]; }
  return null;
};

/**
  * CategoricalList: Renders a list of categorical bin items
  */
class CategoricalList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      expandedBinValue: '',
    };
  }

  getCurrentBinNodes = () => {
    const bin = find(this.props.bins, { value: this.state.expandedBinValue });
    return bin && bin.nodes;
  }

  expandBin = (e, binValue) => {
    e.stopPropagation();
    this.setState({
      expandedBinValue: binValue,
    });
  }

  renderCategoricalBin = (bin, index) => {
    const onDrop = ({ meta }) => {
      if (getNodeAttributes(meta)[this.props.activePromptVariable] === bin.value) {
        return;
      }

      this.props.toggleNodeAttributes(meta[nodePrimaryKeyProperty],
        { [this.props.activePromptVariable]: bin.value });
    };

    const accentColor = getCatColor(index);

    const getDetails = (name) => {
      if (name) {
        if (bin.nodes.length > 0) {
          return `${name}${bin.nodes.length > 1 ? ` and ${bin.nodes.length - 1} other${bin.nodes.length > 2 ? 's' : ''}` : ''}`;
        }
      } else if (bin.nodes.length > 0) {
        return `${bin.nodes.length} node${bin.nodes.length > 1 ? 's' : ''}`;
      }
      return 'empty';
    };

    return (
      <CategoricalItem
        id={`CATBIN_ITEM_${this.props.stage.id}_${this.props.prompt.id}_${index}`}
        key={index}
        label={bin.label}
        accentColor={accentColor}
        onDrop={item => onDrop(item)}
        onClick={e => this.expandBin(e, bin.value)}
        details={name => getDetails(name)}
        isExpanded={this.state.expandedBinValue === bin.value}
        nodes={bin.nodes}
        sortOrder={this.props.prompt.binSortOrder}
      />
    );
  }

  render() {
    const classNames = cx(
      'categorical-list__content',
      { 'categorical-list__content--expanded': this.state.expandedBinValue },
    );

    return (
      <div className="categorical-list" onClick={e => this.expandBin(e, '', '')}>
        <Flipper flipKey={this.state.expandedBinValue}>
          <div
            className={classNames}
            style={{ '--num-categorical-items': this.props.bins.length }}
          >
            {this.props.bins.map(this.renderCategoricalBin)}
          </div>
        </Flipper>
      </div>
    );
  }
}

CategoricalList.propTypes = {
  activePromptVariable: PropTypes.string.isRequired,
  bins: PropTypes.array.isRequired,
  prompt: PropTypes.object.isRequired,
  stage: PropTypes.object.isRequired,
  toggleNodeAttributes: PropTypes.func.isRequired,
};

CategoricalList.defaultProps = {
  isDragging: false,
  meta: {},
};

function makeMapStateToProps() {
  const getCategoricalValues = makeGetVariableOptions();
  const getPromptVariable = makeGetPromptVariable();
  const getStageNodes = makeNetworkNodesForType();

  return function mapStateToProps(state, props) {
    const stageNodes = getStageNodes(state, props);
    const activePromptVariable = getPromptVariable(state, props);

    return {
      activePromptVariable,
      bins: getCategoricalValues(state, props)
        .map((bin) => {
          const nodes = stageNodes.filter(
            node =>
              node[nodeAttributesProperty][activePromptVariable] &&
              node[nodeAttributesProperty][activePromptVariable] === bin.value,
          );

          return {
            ...bin,
            nodes,
          };
        }),
    };
  };
}

function mapDispatchToProps(dispatch) {
  return {
    toggleNodeAttributes: bindActionCreators(sessionsActions.toggleNodeAttributes, dispatch),
  };
}

export default compose(
  connect(makeMapStateToProps, mapDispatchToProps),
  MonitorDragSource(['isDragging', 'meta']),
)(CategoricalList);
