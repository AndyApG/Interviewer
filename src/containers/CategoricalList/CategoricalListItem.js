import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { actionCreators as sessionsActions } from '../../ducks/modules/sessions';
import { CategoricalItem } from '../../components';
import { getEntityAttributes, entityPrimaryKeyProperty } from '../../ducks/modules/network';
import Overlay from '../Overlay';

const formatBinDetails = (nodes, getNodeLabel) => {
  if (nodes.length === 0) {
    return '';
  }

  // todo: the following should be updated to reflect the sort order of the bins
  const name = getNodeLabel(nodes[0]);

  return `${name}${nodes.length > 1 ? ` and ${nodes.length - 1} other${nodes.length > 2 ? 's' : ''}` : ''}`;
};

const CategoricalListItem = ({
  id,
  size,
  isExpanded,
  accentColor,
  activePromptVariable,
  bin,
  index,
  sortOrder,
  getNodeLabel,
  onExpandBin,
  updateNode,
}) => {
  const isOtherVariable = !!bin.otherVariable;
  const binDetails = formatBinDetails(bin.nodes, getNodeLabel);

  const handleDrop = ({ meta }) => {
    const binValue = bin.value;

    if (isOtherVariable) {
      console.log('other variable');
      // this.openOtherFieldWindow();
      return;
    }

    if (getEntityAttributes(meta)[activePromptVariable] === [binValue]) {
      return;
    }

    updateNode(
      meta[entityPrimaryKeyProperty],
      {},
      { [activePromptVariable]: [binValue] },
    );
  };

  const handleExpandBin = (e) => {
    if (e) { e.stopPropagation(); }
    onExpandBin(index);
  };

  return (
    <div
      className="categorical-list__item"
      style={{ width: `${size}px`, height: `${size}px` }}
      key={index}
      onClick={handleExpandBin}
    >
      <CategoricalItem
        id={id}
        key={index}
        label={bin.label}
        accentColor={accentColor}
        onDrop={handleDrop}
        onClick={handleExpandBin}
        details={binDetails}
        isExpanded={isExpanded}
        nodes={bin.nodes}
        sortOrder={sortOrder}
      />
      <Overlay show={false}>
        testing
      </Overlay>
    </div>
  );
};

CategoricalListItem.propTypes = {
  id: PropTypes.string.isRequired,
  isExpanded: PropTypes.bool.isRequired,
  activePromptVariable: PropTypes.string.isRequired,
  bin: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  sortOrder: PropTypes.func.isRequired,
  getNodeLabel: PropTypes.func.isRequired,
  onExpandBin: PropTypes.func.isRequired,
  updateNode: PropTypes.func.isRequired,
  accentColor: PropTypes.string,
  size: PropTypes.number,
};

CategoricalListItem.defaultProps = {
  accentColor: null,
  size: 0,
};

export { CategoricalListItem };

const mapDispatchToProps = {
  updateNode: sessionsActions.updateNode,
};

export default compose(
  connect(null, mapDispatchToProps),
)(CategoricalListItem);
