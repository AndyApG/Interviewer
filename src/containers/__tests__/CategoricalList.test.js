/* eslint-env jest */

import React from 'react';
import { shallow, mount } from 'enzyme';
import { UnconnectedCategoricalList as CategoricalList } from '../CategoricalList';
import { UnconnectedCategoricalItem as CategoricalItem } from '../../components/CategoricalItem';

const mockProps = {
  bins: [{ label: 'one', value: 1, nodes: [1, 2, 3] }, { label: 'two', value: 2, nodes: [4, 5] }],
  stage: { id: 1 },
  prompt: { id: 1 },
};

describe('CategoricalList component', () => {
  it('renders categorical list', () => {
    const component = shallow(
      <CategoricalList {...mockProps} />,
    );

    expect(component).toMatchSnapshot();
  });

  it('items match the number of bins', () => {
    const component = mount(
      <CategoricalList {...mockProps} />,
    );

    expect(component.find(CategoricalItem)).toHaveLength(2);
  });
});
