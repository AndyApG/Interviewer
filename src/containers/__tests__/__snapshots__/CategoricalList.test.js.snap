// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoricalList component renders categorical list 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoricalList
    bins={
      Array [
        Object {
          "label": "one",
          "nodes": Array [
            1,
            2,
            3,
          ],
          "value": 1,
        },
        Object {
          "label": "two",
          "nodes": Array [
            4,
            5,
          ],
          "value": 2,
        },
      ]
    }
    isDragging={false}
    meta={Object {}}
    prompt={
      Object {
        "id": 1,
      }
    }
    stage={
      Object {
        "id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Flipper
        applyTransformOrigin={true}
        element="div"
        flipKey=""
      >
        <div
          className="categorical-list__content"
          style={
            Object {
              "--num-categorical-items": 2,
            }
          }
        >
          <withState(withProps(DropTarget))
            accentColor={null}
            details={[Function]}
            id="CATBIN_ITEM_1_1_0"
            isExpanded={false}
            label="one"
            nodes={
              Array [
                1,
                2,
                3,
              ]
            }
            onClick={[Function]}
            onDrop={[Function]}
            sortOrder={undefined}
          />
          <withState(withProps(DropTarget))
            accentColor={undefined}
            details={[Function]}
            id="CATBIN_ITEM_1_1_1"
            isExpanded={false}
            label="two"
            nodes={
              Array [
                4,
                5,
              ]
            }
            onClick={[Function]}
            onDrop={[Function]}
            sortOrder={undefined}
          />
        </div>
      </Flipper>,
      "className": "categorical-list",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "applyTransformOrigin": true,
        "children": <div
          className="categorical-list__content"
          style={
            Object {
              "--num-categorical-items": 2,
            }
          }
        >
          <withState(withProps(DropTarget))
            accentColor={null}
            details={[Function]}
            id="CATBIN_ITEM_1_1_0"
            isExpanded={false}
            label="one"
            nodes={
              Array [
                1,
                2,
                3,
              ]
            }
            onClick={[Function]}
            onDrop={[Function]}
            sortOrder={undefined}
          />
          <withState(withProps(DropTarget))
            accentColor={undefined}
            details={[Function]}
            id="CATBIN_ITEM_1_1_1"
            isExpanded={false}
            label="two"
            nodes={
              Array [
                4,
                5,
              ]
            }
            onClick={[Function]}
            onDrop={[Function]}
            sortOrder={undefined}
          />
        </div>,
        "element": "div",
        "flipKey": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withState(withProps(DropTarget))
              accentColor={null}
              details={[Function]}
              id="CATBIN_ITEM_1_1_0"
              isExpanded={false}
              label="one"
              nodes={
                Array [
                  1,
                  2,
                  3,
                ]
              }
              onClick={[Function]}
              onDrop={[Function]}
              sortOrder={undefined}
            />,
            <withState(withProps(DropTarget))
              accentColor={undefined}
              details={[Function]}
              id="CATBIN_ITEM_1_1_1"
              isExpanded={false}
              label="two"
              nodes={
                Array [
                  4,
                  5,
                ]
              }
              onClick={[Function]}
              onDrop={[Function]}
              sortOrder={undefined}
            />,
          ],
          "className": "categorical-list__content",
          "style": Object {
            "--num-categorical-items": 2,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "accentColor": null,
              "details": [Function],
              "id": "CATBIN_ITEM_1_1_0",
              "isExpanded": false,
              "label": "one",
              "nodes": Array [
                1,
                2,
                3,
              ],
              "onClick": [Function],
              "onDrop": [Function],
              "sortOrder": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "accentColor": undefined,
              "details": [Function],
              "id": "CATBIN_ITEM_1_1_1",
              "isExpanded": false,
              "label": "two",
              "nodes": Array [
                4,
                5,
              ],
              "onClick": [Function],
              "onDrop": [Function],
              "sortOrder": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Flipper
          applyTransformOrigin={true}
          element="div"
          flipKey=""
        >
          <div
            className="categorical-list__content"
            style={
              Object {
                "--num-categorical-items": 2,
              }
            }
          >
            <withState(withProps(DropTarget))
              accentColor={null}
              details={[Function]}
              id="CATBIN_ITEM_1_1_0"
              isExpanded={false}
              label="one"
              nodes={
                Array [
                  1,
                  2,
                  3,
                ]
              }
              onClick={[Function]}
              onDrop={[Function]}
              sortOrder={undefined}
            />
            <withState(withProps(DropTarget))
              accentColor={undefined}
              details={[Function]}
              id="CATBIN_ITEM_1_1_1"
              isExpanded={false}
              label="two"
              nodes={
                Array [
                  4,
                  5,
                ]
              }
              onClick={[Function]}
              onDrop={[Function]}
              sortOrder={undefined}
            />
          </div>
        </Flipper>,
        "className": "categorical-list",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "applyTransformOrigin": true,
          "children": <div
            className="categorical-list__content"
            style={
              Object {
                "--num-categorical-items": 2,
              }
            }
          >
            <withState(withProps(DropTarget))
              accentColor={null}
              details={[Function]}
              id="CATBIN_ITEM_1_1_0"
              isExpanded={false}
              label="one"
              nodes={
                Array [
                  1,
                  2,
                  3,
                ]
              }
              onClick={[Function]}
              onDrop={[Function]}
              sortOrder={undefined}
            />
            <withState(withProps(DropTarget))
              accentColor={undefined}
              details={[Function]}
              id="CATBIN_ITEM_1_1_1"
              isExpanded={false}
              label="two"
              nodes={
                Array [
                  4,
                  5,
                ]
              }
              onClick={[Function]}
              onDrop={[Function]}
              sortOrder={undefined}
            />
          </div>,
          "element": "div",
          "flipKey": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withState(withProps(DropTarget))
                accentColor={null}
                details={[Function]}
                id="CATBIN_ITEM_1_1_0"
                isExpanded={false}
                label="one"
                nodes={
                  Array [
                    1,
                    2,
                    3,
                  ]
                }
                onClick={[Function]}
                onDrop={[Function]}
                sortOrder={undefined}
              />,
              <withState(withProps(DropTarget))
                accentColor={undefined}
                details={[Function]}
                id="CATBIN_ITEM_1_1_1"
                isExpanded={false}
                label="two"
                nodes={
                  Array [
                    4,
                    5,
                  ]
                }
                onClick={[Function]}
                onDrop={[Function]}
                sortOrder={undefined}
              />,
            ],
            "className": "categorical-list__content",
            "style": Object {
              "--num-categorical-items": 2,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "accentColor": null,
                "details": [Function],
                "id": "CATBIN_ITEM_1_1_0",
                "isExpanded": false,
                "label": "one",
                "nodes": Array [
                  1,
                  2,
                  3,
                ],
                "onClick": [Function],
                "onDrop": [Function],
                "sortOrder": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "accentColor": undefined,
                "details": [Function],
                "id": "CATBIN_ITEM_1_1_1",
                "isExpanded": false,
                "label": "two",
                "nodes": Array [
                  4,
                  5,
                ],
                "onClick": [Function],
                "onDrop": [Function],
                "sortOrder": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
